File: 品味Java的21種設計模式/dp_cpp/singleton/example8/Singleton.cpp
#include "Singleton.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace singleton
			{
				namespace example8
				{

					Singleton::Singleton()
					{
						//
					}

					Singleton Singleton::getInstance()
					{
						//先从缓存中获取
						Singleton instance = static_cast<Singleton>(map->get(DEFAULT_KEY));
						//如果没有，就新建一个，然后设置回缓存中
						if(instance==0)
						{
							instance = new Singleton();
							map->put(DEFAULT_KEY, instance);
						}
						//如果有就直接使用
						return instance;
					}
				}
			}
		}
	}
}

File: ,/japan/Windows/src/Singleton/Q3/Singleton.java
public class Singleton {
    private static Singleton singleton = null;
    private Singleton() {
        System.out.println("已產生實例。");
    }
    public static Singleton getInstance() {
        if (singleton == null) {
            singleton = new Singleton();
        }
        return singleton;
    }
}
